### DISCUSSION TITLE
Is it possible to use Pytest BDD without writing feature file ?

### DISCUSSION THREAD

**Discussion by jossefaz** [pytest-dev/pytest-bdd#603] [Q&A]
Is it possible to use pytest-bdd features (like the "given" decorator) without writing gherkin ".feature" file, but just based on list of strings ?

For exmple I have this list as an input :
```python
some_steps = ["A user with roles X,Y,Z", "Click on this button"]
```

Now I would like to execute those steps using a simple for loop that iterates through the `some_steps` variable. 
Let's say for instance that I expect those functions to be called : 

```python
@given("A user with {roles}", target_fixture="user")
def user_precondition():
     # some logic to create a user here

@when("Click on this button")
def user_click(user):
    # some logic here
```

**Comment by youtux** [comment]
I'm sorry but this is not supported out of the box.

The only workaround I can think of is to make your own function that generates scenario files programmatically, and call it at import time in your test module.

**Reply by evbo** [reply]
@youtux it's hard to discern in this issue, but it appears long ago there was work done to generate one from the other: https://github.com/pytest-dev/pytest-bdd/issues/103

I just learned about this repo, but as an outsider it seems in the best interest of everyone to automate the creation of `feature` files from `test` files, to avoid redundant labor and bugs due to them falling out of sync. I'm just surprised this isn't a more popular feature request. Any reasons not to support this sort of automation?

**Reply by evbo** [reply]
so it looks like you can't generate `.feature` files automatically, but you can generate test code automatically... I wish you could generate either..

https://github.com/pytest-dev/pytest-bdd/issues/32
